api/client/service/opts.go

```diff
@@ -253,6 +307,21 @@ func (m *MountOpt) Value() []swarm.Mount {
 	return m.values
 }
 
+func getMountType(fields []string) (swarm.MountType, []string) {
```

Comment:
@dnephin And we've already made compromises to make the syntax easier. Now, you've gone off an editorialized the grammar without giving it a proper analysis and I can't tell if we are introducing the same issues or not. What was simple with a few rewrites now has keys and values that are only valid in certain branches of the parser. It also makes additions and extensions more challenging.

Look, I appreciate that you've taken the time to submit this PR and address the issues identified, but this is going too far and making the valid cases too complex. The previous version reflected the fact that mounts share structure. 

__Now that these don't even share the same key space, they might as well be completely different flags, as they no long share any structure.__